src/utils/forecast.js

```diff
@@ -1,21 +1,18 @@
-const request = require('request')
+const request = require('request-promise-native')
 
-const forecast = (lat, lon, callback) => {
-    const url = `http://api.weatherstack.com/current?access_key=4b2554510856f2dcc9ef34f90a838eb9&query=${encodeURIComponent(lat)},${encodeURIComponent(lon)}&units=m`
-    request({ url, json: true }, (error, { body }) => {
-        if (error) {
-            callback('No connection available.', undefined)
-        } else if (body.error) {
-            callback('No match found.', undefined)
-        } else {
-            callback(undefined, {
-                description: body.current.weather_descriptions[0],
-                temperature: body.current.temperature,
-                feelslike: body.current.feelslike,
-                icon: body.current.weather_icons[0]
-            })
+const forecast = async ({ latitude, longitude }) => {
+    const url = `http://api.weatherstack.com/current?access_key=4b2554510856f2dcc9ef34f90a838eb9&query=${encodeURIComponent(latitude)},${encodeURIComponent(longitude)}&units=m`
```

Comment:
access keys must not be public and should be located in .env files

also, url generation must not located here

I suggest you to use entities and repositories patter, i will prepare an example and we discuss that on next call

Reply:
Moved access key into a separate file.